Fix JavaScript version

### Have you experienced this bug with the latest version of the template?

Yes

### Steps to Reproduce

Create a new project with this template and choose JavaScript.

### Expected Behavior

No errors when running `npm run validate`

### Actual Behavior

After @MichaelDeBoey's PR #127 it looks like there are still more fixes to be made.

I ran into a couple issues when running `npm run validate`: 

```
TypeError: tsconfigPaths is not a function
```

After fixing that, I got:

```

/Users/kentcdodds/Desktop/my-remix-app/app/routes/notes/$noteId.jsx
  6:1  warning  '~/models/note.server' import is duplicated  no-duplicate-imports

/Users/kentcdodds/Desktop/my-remix-app/cypress/e2e/smoke.cy.js
   3:1  error  'describe' is not defined   no-undef
   4:3  error  'afterEach' is not defined  no-undef
   5:5  error  'cy' is not defined         no-undef
   8:3  error  'it' is not defined         no-undef
  14:5  error  'cy' is not defined         no-undef
  16:5  error  'cy' is not defined         no-undef
  18:5  error  'cy' is not defined         no-undef
  20:5  error  'cy' is not defined         no-undef
  21:5  error  'cy' is not defined         no-undef
  22:5  error  'cy' is not defined         no-undef
  24:5  error  'cy' is not defined         no-undef
  25:5  error  'cy' is not defined         no-undef
  26:5  error  'cy' is not defined         no-undef
  29:3  error  'it' is not defined         no-undef
  35:5  error  'cy' is not defined         no-undef
  37:5  error  'cy' is not defined         no-undef
  39:5  error  'cy' is not defined         no-undef
  40:5  error  'cy' is not defined         no-undef
  42:5  error  'cy' is not defined         no-undef
  44:5  error  'cy' is not defined         no-undef
  45:5  error  'cy' is not defined         no-undef
  46:5  error  'cy' is not defined         no-undef
  48:5  error  'cy' is not defined         no-undef
  50:5  error  'cy' is not defined         no-undef

/Users/kentcdodds/Desktop/my-remix-app/cypress/support/commands.js
   6:3   error  'cy' is not defined                                                  no-undef
   7:3   error  'cy' is not defined                                                  no-undef
  13:5   error  'cy' is not defined                                                  no-undef
  15:10  error  'cy' is not defined                                                  no-undef
  22:5   error  'cy' is not defined                                                  no-undef
  29:3   error  'cy' is not defined                                                  no-undef
  33:3   error  'cy' is not defined                                                  no-undef
  37:3   error  'cy' is not defined                                                  no-undef
  46:3   error  'cy' is not defined                                                  no-undef
  47:3   error  'cy' is not defined                                                  no-undef
  50:1   error  'Cypress' is not defined                                             no-undef
  51:1   error  'Cypress' is not defined                                             no-undef
  52:1   error  'Cypress' is not defined                                             no-undef
  54:1   error  Definition for rule '@typescript-eslint/no-namespace' was not found  @typescript-eslint/no-namespace

/Users/kentcdodds/Desktop/my-remix-app/cypress/support/e2e.js
  4:1  error  'Cypress' is not defined  no-undef

✖ 40 problems (39 errors, 1 warning)
```

Running `npm run test:e2e:run` I got:

```
Your configFile is invalid: /Users/kentcdodds/Desktop/my-remix-app/cypress.config.js

It threw an error when required, check the stack trace below:

/Users/kentcdodds/Desktop/my-remix-app/cypress.config.js:3
export default defineConfig({
^^^^^^

SyntaxError: Unexpected token 'export'
    at Object.compileFunction (node:vm:353:18)
    at wrapSafe (node:internal/modules/cjs/loader:1040:15)
    at Module._compile (node:internal/modules/cjs/loader:1076:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1166:10)
    at Module.load (node:internal/modules/cjs/loader:988:32)
    at Module._load (node:internal/modules/cjs/loader:834:12)
    at Module.require (node:internal/modules/cjs/loader:1012:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at loadFile (/Users/kentcdodds/Library/Caches/Cypress/10.4.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/child/run_require_async_child.js:89:14)
    at EventEmitter.<anonymous> (/Users/kentcdodds/Library/Caches/Cypress/10.4.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/child/run_require_async_child.js:116:38)
    at EventEmitter.emit (node:events:537:28)
    at process.<anonymous> (/Users/kentcdodds/Library/Caches/Cypress/10.4.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/util.js:33:22)
    at process.emit (node:events:537:28)
    at emit (node:internal/child_process:937:14)
    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)

Error: Command failed with exit code 1: npx cypress run
    at makeError (/Users/kentcdodds/Desktop/my-remix-app/node_modules/start-server-and-test/node_modules/execa/lib/error.js:60:11)
    at handlePromise (/Users/kentcdodds/Desktop/my-remix-app/node_modules/start-server-and-test/node_modules/execa/index.js:118:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  shortMessage: 'Command failed with exit code 1: npx cypress run',
  command: 'npx cypress run',
  escapedCommand: '"npx cypress run"',
  exitCode: 1,
  signal: undefined,
  signalDescription: undefined,
  stdout: undefined,
  stderr: undefined,
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
}
NPM command npm run test:e2e:run -- failed with code 1
```

--- a/.eslintrc.js
+++ b/.eslintrc.js
@@ -8,6 +8,10 @@ module.exports = {
     "@remix-run/eslint-config/jest-testing-library",
     "prettier",
   ],
+  env: {
+    "cypress/globals": true,
+  },
+  plugins: ["cypress"],
   // we're using vitest which has a very similar API to jest
   // (so the linting plugins work nicely), but it means we have to explicitly
   // set the jest version.

--- a/cypress/support/commands.ts
+++ b/cypress/support/commands.ts
@@ -93,8 +93,3 @@ function visitAndCheck(url: string, waitTime: number = 1000) {
 Cypress.Commands.add("login", login);
 Cypress.Commands.add("cleanupUser", cleanupUser);
 Cypress.Commands.add("visitAndCheck", visitAndCheck);
-
-/*
-eslint
-  @typescript-eslint/no-namespace: "off",
-*/

--- a/package.json
+++ b/package.json
@@ -63,6 +63,7 @@
     "cypress": "^10.4.0",
     "eslint": "^8.21.0",
     "eslint-config-prettier": "^8.5.0",
+    "eslint-plugin-cypress": "^2.12.1",
     "happy-dom": "^6.0.4",
     "msw": "^0.44.2",
     "npm-run-all": "^4.1.5",

--- a/remix.init/index.js
+++ b/remix.init/index.js
@@ -10,9 +10,9 @@ const YAML = require("yaml");
 const cleanupCypressFiles = (filesEntries) =>
   filesEntries.flatMap(([filePath, content]) => {
     const newContent = content
-      .replace("npx ts-node", "node")
-      .replace("create-user.ts", "create-user.js")
-      .replace("delete-user.ts", "delete-user.js");
+      .replace(new RegExp("npx ts-node", "g"), "node")
+      .replace(new RegExp("create-user.ts", "g"), "create-user.js")
+      .replace(new RegExp("delete-user.ts", "g"), "delete-user.js");
 
     return [fs.writeFile(filePath, newContent)];
   });
@@ -46,9 +46,9 @@ const readFileIfNotTypeScript = (
   filePath,
   parseFunction = (result) => result
 ) =>
-  (isTypeScript ? Promise.resolve() : fs.readFile(filePath, "utf-8")).then(
-    parseFunction
-  );
+  isTypeScript
+    ? Promise.resolve()
+    : fs.readFile(filePath, "utf-8").then(parseFunction);
 
 const removeUnusedDependencies = (dependencies, unusedDependencies) =>
   Object.fromEntries(
@@ -68,12 +68,9 @@ const updatePackageJson = ({ APP_NAME, isTypeScript, packageJson }) => {
     name: APP_NAME,
     devDependencies: isTypeScript
       ? devDependencies
-      : removeUnusedDependencies(devDependencies, [
-          "ts-node",
-          "vite-tsconfig-paths",
-        ]),
+      : removeUnusedDependencies(devDependencies, ["ts-node"]),
     prisma: isTypeScript
-      ? prisma
+      ? { ...prisma, seed: prismaSeed }
       : {
           ...prisma,
           seed: prismaSeed

